### PR TITLE
Add additional range params

### PR DESCRIPTION
**Pull request by bion**
- [x] add number inputs for remaining continuous(ish) params
- [x] map MIDI param values (0-127) to display values (0-10) and back

### CODE REVIEW

**Comment by bion** [comment]
screen shot 2016-01-12 at 12 06 43 am

**Comment by bion** [comment]
@dbalatero

File: src/js/util/param_spec.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+function ParamSpec(sourceLow, sourceHigh, distLow, distHigh, clamp = true) {
+  const domain = sourceHigh - sourceLow
+  const range = distHigh - distLow
+  let map
+  let unmap
+
+  if (clamp) {
+    map = (val) => {
```
Comment (dbalatero) [inline src/js/util/param_spec.js]: map/unmap aren't really clear function names to me?

File: src/js/util/param_spec.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+function ParamSpec(sourceLow, sourceHigh, distLow, distHigh, clamp = true) {
```
Comment (dbalatero) [inline src/js/util/param_spec.js]: Is there a better name for this? I don't really know what the `Spec` part conveys?

File: src/js/util/param_spec.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  const domain = sourceHigh - sourceLow
+  const range = distHigh - distLow
+  let map
+  let unmap
+
+  if (clamp) {
```
Comment (dbalatero) [inline src/js/util/param_spec.js]: Do we use `clamp`? If no, why add it?

Comment (bion) [inline src/js/util/param_spec.js]: > Do we use clamp? If no, why add it?

yeah, it clamps by default which is how it's used [here](https://github.com/morons-llc/synth-frontend/pull/4/files#diff-5c0e44c2d96f70060565edab9aed2d62R6)

File: src/js/util/param_spec.js (hunk 2 above)
Comment (bion) [inline src/js/util/param_spec.js]: I was thinking of this as mapping or specifying a range of values for a given parameter. Maybe we could call it `RangeMapper`?

> map/unmap aren't really clear function names to me?

if we called it `RangeMapper` would map/unmap work? names are hard.

File: src/js/util/param_spec.js (hunk 3 above)
Comment (dbalatero) [inline src/js/util/param_spec.js]: what is clamping?

File: test/util/param_spec_test.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    describe('map', function () {
+      it('maps a value to the distribution', function () {
+        expect(this.subject.map(63.5)).to.eql(5)
+      })
+
+      it('maps a value to the distribution', function () {
+        expect(this.subject.map(0)).to.eql(0)
+      })
+
+      it('maps a value to the distribution', function () {
```
Comment (dbalatero) [inline test/util/param_spec_test.js]: Why does 127 become 10? sorry i'm just really confused!

Don't we just have sliders from 0 - 127? What is distribution for?

File: src/js/util/param_spec.js (hunk 3 above)
Comment (bion) [inline src/js/util/param_spec.js]: if asked to map a value out of range it restricts to the min/max of the output range

Comment (bion) [inline src/js/util/param_spec.js]: e.g.

``` javascript
let spec = ParamSpec(0, 10, 0, 100) // clamping on by default
spec.map(5) // => 50
spec.map(10) // => 100
spec.map(11) // => 100
spec.map(-1) // => 0
```

File: test/util/param_spec_test.js (hunk 4 above)
Comment (bion) [inline test/util/param_spec_test.js]: it's to keep our users from getting confused about the numbers they see in the UI. the Juno106 sliders are labeled from 0 - 10

Comment (dbalatero) [inline test/util/param_spec_test.js]: ohhh i see

**Comment by bion** [comment]
@dbalatero name changed, merge?

**Comment by dbalatero** [comment]
go for it!

**Pull request merged**